src/acp/rule.ts

```diff
@@ -233,3 +247,257 @@ export function setForbiddenRuleOnPolicy(policy: Policy, rule: Rule): Policy {
+ * function is still experimental and subject to change, even in a non-major release.
+ * ```
+ *
+ * Prevent the [[Rule]] from applying to a given group.
+ *
+ * @param rule The [[Rule]] that should no longer apply to a given group.
+ * @param agent The group the rule should no longer apply to.
+ * @returns A copy of the rule, not applying to the given group.
+ * @since Unreleased
+ */
+export function removeGroupFromRule(rule: Rule, group: UrlString): Rule {
+  return removeIri(rule, acp.group, group);
+}
+
+/**
+ * ```{note} There is no Access Control Policies specification yet. As such, this
+ * function is still experimental and subject to change, even in a non-major release.
+ * ```
+ *
+ * Check if the rule applies to any agent.
+ *
+ * @param rule The rule checked for public access.
+ * @returns Whether the rule applies to any agent or not.
+ */
+export function hasPublicInRule(rule: Rule): boolean {
+  return (
+    getIriAll(rule, acp.agent).filter((agent) => agent === acp.PublicAgent)
+      .length > 0
+  );
+}
+
+/**
+ * ```{note} There is no Access Control Policies specification yet. As such, this
+ * function is still experimental and subject to change, even in a non-major release.
+ * ```
+ *
+ * Enable or disable a rule from applying to any agent.
+ *
+ * @param rule The rule being modified.
+ * @param hasPublic A boolean indicating whether the rule should apply or not to any agent.
+ * @returns A copy of the rule, updated to apply/not apply to any agent.
+ * @status Unreleased
+ */
+export function setPublicForRule(rule: Rule, hasPublic: boolean): Rule {
+  if (!hasPublic) {
+    return removeIri(rule, acp.agent, acp.PublicAgent);
+  }
+  return addIri(rule, acp.agent, acp.PublicAgent);
+}
+
+/**
+ * ```{note} There is no Access Control Policies specification yet. As such, this
+ * function is still experimental and subject to change, even in a non-major release.
+ * ```
+ *
+ * Check if the rule applies to any authenticated agent.
+ *
+ * @param rule The rule checked for authenticated access.
+ * @returns Whether the rule applies to any authenticated agent or not.
+ */
+export function hasAuthenticatedInRule(rule: Rule): boolean {
+  return (
+    getIriAll(rule, acp.agent).filter(
+      (agent) => agent === acp.AuthenticatedAgent
+    ).length > 0
+  );
+}
+
+/**
+ * ```{note} There is no Access Control Policies specification yet. As such, this
+ * function is still experimental and subject to change, even in a non-major release.
+ * ```
+ *
+ * Enable or disable a rule from applying to any authenticated agent.
+ *
+ * @param rule The rule being modified.
+ * @param hasPublic A boolean indicating whether the rule should apply or not to any authenticated agent.
+ * @returns A copy of the rule, updated to apply/not apply to any authenticated agent.
+ * @status Unreleased
+ */
+export function setAuthenticatedForRule(
+  rule: Rule,
+  authenticated: boolean
+): Rule {
+  if (!authenticated) {
```

Comment:
Being very pedantic, but on simple 'if' statements like this, I tend to favour testing positive conditions not negative (i.e. just remove the 'not' and reverse the clauses). And for conditionals **_this simple_**, I'd just use the ternary operator:
```
return authenticated : addIri(rule, acp.agent, acp.AuthenticatedAgent) : removeIri(rule, acp.agent, acp.AuthenticatedAgent);
```

Reply:
Fixed as part of 9107fe466f0f0d056bdb8146e6e14f59347f0213